flare/icons.py

```diff
@@ -64,15 +64,11 @@ def getIcon(self):
             )
 
     def replaceSVG(self, icondata):
+        classes = list(self["class"])  # We need to restore our classes later
         conf["flare.icon.cache"].update({self.url:icondata})
-        self.removeAllChildren()
-
-        for node in html5.fromHTML(icondata):
-            if isinstance(node, html5.svg.Svg):
-                self["viewbox"] = node["viewbox"]
-                self.addClass(node["class"])
-                self.appendChild(node._children)
-                break
+        self.element.outerHTML = icondata  # Replace our HTML-Code with the contents of the SVG
```

Comment:
Here is a suggestion that would work, but this isn't nice.
```suggestion
        parser = html5.jseval("new DOMParser")
        doc = parser.parseFromString(icondata, "image/svg+xml")
        self.element.replaceWith(doc.firstChild)
        self.element = doc.firstChild
```